packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -207,6 +213,19 @@ export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
         margin-top: 0; /* The margin effectively comes from the top: -2px on the label above it */
       `
     ),
+    invalid: css`
+      input + span,
+      input:checked + span,
```

Comment:
we would need to add another `input + span:hover` to get it to be red all the time. This is a lot of checks though, quite a bit prone to failure. I wonder if we can figure it out programatically what the color of the border would be, like pass in the props for invalid/checked/indeterminate and determine the correct color instead of having all these overrides...

Reply:
I deleted `stylesFactory` as it is deprecated and added a border to all possible states. This border changes its color to 'error' when it is invalid, otherwise, it stays as it is. What do you think? 🤔